### PR TITLE
Add mentor to admin panel

### PR DESCRIPTION
**Pull request by Vlad1023**
Temporary without Mentor`s student groups and courses ( will do in separate pull request )

### CODE REVIEW

**Review by SnipeFrost — APPROVED** [review]

File: CharlieBackend.Api/Controllers/MentorsController.cs [hunk 1]
```diff
@@ -48,6 +49,41 @@ public async Task<ActionResult> PostMentor(long accountId)
...
 
+        /// <summary>
+        /// Get only active students
+        /// </summary>
+        /// <response code="200">Successful return of students list</response>
+        [Authorize(Roles = "Admin, Mentor, Secretary")]
+        [HttpGet("active")]
+        public async Task<ActionResult<IList<MentorDto>>> GetAllActiveStudents()
+        {
+
```
Comment (SerjKocherga) [inline CharlieBackend.Api/Controllers/MentorsController.cs:60]: Remove this empty line.

File: CharlieBackend.Api/Controllers/MentorsController.cs [hunk 2]
```diff
@@ -48,6 +49,41 @@ public async Task<ActionResult> PostMentor(long accountId)
...
+        /// Get mentor information by mentor id
+        /// </summary>
+        /// <response code="200">Successful return of mentor</response>
+        /// <response code="409">Error, can not find mentor</response>
+        [SwaggerResponse(200, type: typeof(MentorMock))]
+        [Authorize(Roles = "Admin, Mentor, Secretary")]
+        [HttpGet("{id}")]
+        public async Task<ActionResult<MentorDto>> GetMentorById(long id)
+        {
+
```
Comment (SerjKocherga) [inline CharlieBackend.Api/Controllers/MentorsController.cs:76]: Remove this empty line.

File: CharlieBackend.Business/Services/Interfaces/IMentorService.cs [hunk 3]
```diff
@@ -9,6 +9,7 @@ namespace CharlieBackend.Business.Services.Interfaces
     public interface IMentorService
     {
         public Task<Result<MentorDto>> CreateMentorAsync(long accountId);
+        Task<Result<IList<MentorDto>>> GetAllActiveMentorsAsync();
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/Interfaces/IMentorService.cs:12]: Add a new line before this method.

File: CharlieBackend.Data/Repositories/Impl/Interfaces/IMentorRepository.cs [hunk 4]
```diff
@@ -14,6 +14,8 @@ public interface IMentorRepository : IRepository<Mentor>
         void UpdateMentorGroups(IEnumerable<MentorOfStudentGroup> currentItems,
                                 IEnumerable<MentorOfStudentGroup> newItems);
 
+        Task<List<Mentor>> GetAllActiveAsync();
+
         //public Task<Mentor> GetAccountByMentorIdAsync(long mentorId);
```
Comment (SerjKocherga) [inline CharlieBackend.Data/Repositories/Impl/Interfaces/IMentorRepository.cs:19]: What this commentary are doing here? Remove it.

**Review by SerjKocherga — CHANGES_REQUESTED** [review]

**Review by sober607 — APPROVED** [review]

**Review by x1mx1x — APPROVED** [review]

File: CharlieBackend.Api/Controllers/MentorsController.cs [hunk 5]
```diff
@@ -48,6 +49,41 @@ public async Task<ActionResult> PostMentor(long accountId)
...
+            var mentorModel = await _mentorService.GetMentorByIdAsync(id);
+
+            if (mentorModel != null)
+            {
+                return Ok(mentorModel);
+            }
+
+            return StatusCode(409, "Cannot find mentor with such id.");
```
Comment (whereismyroot) [inline CharlieBackend.Api/Controllers/MentorsController.cs:84]: Use NotFound method

File: CharlieBackend.Api/Controllers/MentorsController.cs [hunk 6]
```diff
@@ -48,6 +49,41 @@ public async Task<ActionResult> PostMentor(long accountId)
...
+            var mentors = await _mentorService.GetAllActiveMentorsAsync();
+
+            return mentors.ToActionResult();
+        }
+
+        /// <summary>
+        /// Get mentor information by mentor id
+        /// </summary>
+        /// <response code="200">Successful return of mentor</response>
+        /// <response code="409">Error, can not find mentor</response>
```
Comment (whereismyroot) [inline CharlieBackend.Api/Controllers/MentorsController.cs:70]: use 404 code when resource was not found

File: CharlieBackend.Api/SwaggerExamples/MentorsController/GetStudentByIdResponse.cs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+            return new MentorMock
+            {
+                First_name = "Brad",
+                Last_name = "Pitt",
+                Email = "student@example.com"
+            };
+        }
+    }
+
+    class MentorMock
```
Comment (whereismyroot) [inline CharlieBackend.Api/SwaggerExamples/MentorsController/GetStudentByIdResponse.cs:22]: Why do you need this class if GetMentorById returns MentorDto?

**Review by whereismyroot — CHANGES_REQUESTED** [review]

File: CharlieBackend.AdminPanel/Controllers/MentorsController.cs [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+        [HttpPost("{id}")]
+        public async Task<IActionResult> UpdateMentor(long id, UpdateMentorDto data)
+        {
+            var updatedStudent = await _mentorService.UpdateMentorAsync(id, data, Request.Cookies["accessToken"]);
+
+            return RedirectToAction("AllMentors", "Mentors");
+        }
+
+        [HttpPost]
```
[the remark this answers is not in the document]
Comment (Vlad1023) [reply CharlieBackend.AdminPanel/Controllers/MentorsController.cs:49]: If you mean AddMentor endpoint, there`s a function`s parameter

File: CharlieBackend.Api/Controllers/MentorsController.cs (hunk 5 above)
Comment (Vlad1023) [reply CharlieBackend.Api/Controllers/MentorsController.cs:84]: done

File: CharlieBackend.Api/Controllers/MentorsController.cs (hunk 6 above)
Comment (Vlad1023) [reply CharlieBackend.Api/Controllers/MentorsController.cs:70]: done

File: CharlieBackend.Api/SwaggerExamples/MentorsController/GetStudentByIdResponse.cs (hunk 7 above)
Comment (Vlad1023) [reply CharlieBackend.Api/SwaggerExamples/MentorsController/GetStudentByIdResponse.cs:22]: changed to MentorDto

**Review by SerjKocherga — APPROVED** [review]

**Review by BohdanBBG — APPROVED** [review]

**Review by SnipeFrost — APPROVED** [review]

**Review by BohdanBBG — APPROVED** [review]

**Review by whereismyroot — APPROVED** [review]

**Pull request merged**